shell/platform/windows/direct_manipulation.cc

```diff
@@ -53,16 +56,31 @@ HRESULT DirectManipulationEventHandler::OnViewportStatusChanged(
             (int32_t) reinterpret_cast<int64_t>(this));
       }
     }
-  } else if (previous == DIRECTMANIPULATION_RUNNING) {
+  }
+  if (previous == DIRECTMANIPULATION_RUNNING) {
```

Comment:
```suggestion
  else if (previous == DIRECTMANIPULATION_RUNNING) {
    if (during_synthesized_reset) {
    during_synthesized_reset_ = false;
  } 
```

And remove L 49-50 might make the logic more readable.